### PR TITLE
Return iso8601 values for DEGSTDT, DEGENDDT in HESA export

### PR DESCRIPTION
**Pull request by mnacos**
## Context

HESA requires dates to follow the ISO format, however we don't collect the day/month for degree start and end dates from candidates. We currently only return YYYY in the export. We should return what we have as YYYY-01-01 to help providers stay compliant when preparing their census submissions.

## Changes proposed in this pull request

No UI changes.

## Guidance to review

Run the specs.

## Link to Trello card

https://trello.com/c/LPViuJit

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/decorators/application_choice_export_decorator.rb [hunk 1]
```diff
@@ -24,6 +24,12 @@ def degrees_completed_flag
     application_form.degrees_completed ? 1 : 0
   end
 
+  def first_degree
```
Comment (duncanjbrown) [inline app/decorators/application_choice_export_decorator.rb:27]: :+1:

**Review by duncanjbrown — COMMENTED** [review]
Looking good! Can we be more confident about the presence of a first degree (ie fewer `&`s) as it's a hard requirement for submitting the form?

**Comment by mnacos** [comment]
> Looking good! Can we be more confident about the presence of a first degree (ie fewer `&`s) as it's a hard requirement for submitting the form?

Ok, le'ts do that and find out if any exports break. I've updated all `ApplicationChoice` factories to create applications with degrees.

File: app/services/provider_interface/hesa_data_export.rb [hunk 2]
```diff
@@ -99,6 +98,14 @@ def qualification_aim(application)
 
       '021'
     end
+
+    def first_degree_year(application, year_type)
+      application.first_degree&.send year_type
```
Comment (duncanjbrown) [inline app/services/provider_interface/hesa_data_export.rb:103]: Could lose the `&` here?

**Review by duncanjbrown — APPROVED** [review]
Nice — good to be bold!

**Pull request merged**